sklearn/linear_model/tests/test_sparse_coordinate_descent.py

```diff
@@ -374,3 +375,17 @@ def test_sparse_enet_coordinate_descent():
     )
     with pytest.warns(ConvergenceWarning, match=warning_message):
         clf.fit(X, y)
+
+
+def test_sparse_read_only_buffer():
+    """Test that sparse coordinate descent works for read-only buffers"""
+    rng = np.random.RandomState(0)
+
+    clf = ElasticNet(alpha=0.1, copy_X=False, random_state=rng)
```

Comment:
Does this require `copy_X=False` or does it work with both True and False?

Reply:
Expanded the test to cover both cases -- thanks!